hbase-client/src/main/java/org/apache/hadoop/hbase/quotas/QuotaRetriever.java

```diff
@@ -156,26 +137,26 @@ public void remove() {
 
   /**
    * Open a QuotaRetriever with no filter, all the quota settings will be returned.
-   * @param conf Configuration object to use.
+   * @param conn Connection object to use.
    * @return the QuotaRetriever
    * @throws IOException if a remote or network exception occurs
    */
-  public static QuotaRetriever open(final Configuration conf) throws IOException {
-    return open(conf, null);
+  public static QuotaRetriever open(final Connection conn) throws IOException {
```

Comment:
Because this is `IA.Public`, you cannot make these blanket changes in one pass. You'll need to observe a deprecation cycle through a major release in order to make breaking changes to the public API.

We document this in depth over on https://hbase.apache.org/book.html#hbase.versioning

If we're going through the trouble to make breaking changes, let's push RAII and do away with the parameterless constructor + init method.

Reply:
+1 for the deprecation cycle.

Reply:
> If we're going through the trouble to make breaking changes, let's push RAII and do away with the parameterless constructor + init method.

Since we're opening this worm-can, how about we get rid of these static constructor methods and use constructors like a normal object?